### PR TITLE
[wip] Move libethcore/Transaction to libethcore/TransactionBase

### PR DESCRIPTION
**Pull request by pirapira**
because it's about class TransactionBase, not Transaction. There is also libethereum/Transaction.h

### CODE REVIEW

**Comment by pirapira** [comment]
I also noticed a local variable that is never read.

File: libethcore/TransactionBase.cpp [hunk 1]
```diff
@@ -46,28 +46,27 @@ TransactionBase::TransactionBase(TransactionSkeleton const& _ts, Secret const& _
 
 TransactionBase::TransactionBase(bytesConstRef _rlpData, CheckTransaction _checkSig)
 {
-	int field = 0;
-	RLP rlp(_rlpData);
+	RLP const rlp(_rlpData);
```
Comment (gumb0) [inline libethcore/TransactionBase.cpp]: `RLP` class is immutable, so usually I don't bother declaring it const

Comment (pirapira) [reply libethcore/TransactionBase.cpp]: I see.  I didn't know that.

**Comment by chfast** [comment]
I'm not sure I didn't also want to merge Transaction from libethereum with this one. Or other way around.

**Comment by pirapira** [comment]
I'll merge to `libethereum/Transaction.h`.

**Comment by pirapira** [comment]
Merging them involves a big operation.  Either
* move `libethereum/Interaface` to `libethcore` or
* move `libethcore/SealEngine` to `libethereum` is necessary.

**Pull request merged**